domain/application/service/application.go

```diff
@@ -1464,6 +1478,42 @@ func (s *Service) GetApplicationConstraints(ctx context.Context, appID coreappli
 	return cons, internalerrors.Capture(err)
 }
 
+// GetApplicationStatus looks up the status of the specified application,
+// returning an error satisfying [applicationerrors.ApplicationNotFound] if the
+// application is not found.
+func (s *Service) GetApplicationStatus(ctx context.Context, appID coreapplication.ID) (*corestatus.StatusInfo, error) {
```

Comment:
appId vs appName.

In the case of units, we take in a unit name and convert to an id in flight. Is it natural for these methods to be called with appIds?

Reply:
Yes, we'll need to make another round of changes to fix this. Not here, we're just being consistent.